DSCResources/MSFT_xDnsServerPrimaryZone/MSFT_xDnsServerPrimaryZone.psm1

```diff
@@ -0,0 +1,177 @@
+Import-Module $PSScriptRoot\..\Helper.psm1 -Verbose:$false
+
+# Localized messages
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData @'
+CheckingZoneMessage          = Checking DNS server zone with name '{0}' is '{1}'...
+AddingZoneMessage            = Adding DNS server zone '{0}' ...
+RemovingZoneMessage          = Removing DNS server zone '{0}' ...
+
+CheckPropertyMessage         = Checking DNS server zone property '{0}' ...
+NotDesiredPropertyMessage    = DNS server zone property '{0}' is not correct. Expected '{1}', actual '{2}'
+SetPropertyMessage           = DNS server zone property '{0}' is set
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$Name,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$ZoneFile,
+
+        [ValidateSet('None','NonsecureAndSecure')]
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    $targetResource = @{
+        Name = $Name;
+        ZoneFile = $ZoneFile;
+        DynamicUpdate = $DynamicUpdate;
+        Ensure = ''
+    }
+    if ($Ensure -eq 'Present')
+    {
+        if (Test-TargetResource @PSBoundParameters)
+        {
+            $targetResource['Ensure'] = 'Present';
+        }
+        else {
+            $targetResource['Ensure'] = 'Absent';
+        }
+    }
+    elseif ($Ensure -eq 'Absent')
+    {
+        if (Test-TargetResource @PSBoundParameters)
+        {
+            $targetResource['Ensure'] = 'Absent';
+        }
+        else
+        {
+            $targetResource['Ensure'] = 'Present';
+        }
+    }
+
+    return $targetResource;
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$Name,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]$ZoneFile,
+
+        [ValidateSet('None','NonsecureAndSecure')]
+        [System.String]$DynamicUpdate = 'None',
+        
+        [ValidateSet('Present','Absent')]
+        [System.String]$Ensure = 'Present'
+    )
+
+    Assert-Module -ModuleName 'DNSServer';
+    Write-Verbose ($LocalizedData.CheckingZoneMessage -f $Name, $Ensure);
+    $dnsServerZone = Get-DnsServerZone -Name $Name -ErrorAction SilentlyContinue;
```

Comment:
I would prefer that this code goes in Get-TargetResource function which should return the current state of PrimaryZone